### ISSUE TITLE
Wrong "warning: subtraction of pointers to type 'int [n]' of zero size has undefined behavior [-Wpointer-arith]"

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#28328]
# Bug 27954: Incorrect warning for pointer subtraction on VLA type

| Field | Details |
|---|---|
| Bugzilla Link | [27954](https://llvm.org/bz27954) |
| Version | trunk |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |

### Source code

```c
#include <stdio.h>

int main()
{
  int n = 1;
  int a[n];
  (void)(&a + 1 - &a);

  printf("sizeof(int [n]) = %zu\n", sizeof(int [n]));
}
```

### Results

```text
$ clang -std=c11 -Weverything -Wno-vla -O3 test.c && ./a.out
test.c:7:17: warning: subtraction of pointers to type 'int [n]' of zero size has undefined behavior [-Wpointer-arith]
  (void)(&a + 1 - &a);
         ~~~~~~ ^ ~~
1 warning generated.

sizeof(int [n]) = 4
```

The warning talks about zero size, but the size of the type `int [n]` is not zero, as demonstrated by the output:

```text
sizeof(int [n]) = 4
```

The warning therefore appears to be incorrect.

### Clang version

```text
clang version 3.9.0 (trunk 271312)
```
